apis/clusters/v1beta1/validation.go

```diff
@@ -392,3 +392,31 @@ func (s *GenericDataCentreSpec) validateCreation() error {
 
 	return nil
 }
+
+func (s *GenericDataCentreSpec) ValidateOnPremisesCreation() error {
+	if s.CloudProvider != models.ONPREMISES {
+		return fmt.Errorf("cloud provider %s is unavailable for data centre: %s, available value: %s",
+			s.CloudProvider, s.Name, models.ONPREMISES)
+	}
+
+	if s.Region != models.CLIENTDC {
+		return fmt.Errorf("region %s is unavailable for data centre: %s, available value: %s",
+			s.Region, s.Name, models.CLIENTDC)
+	}
```

Comment:
Maybe it is better to make it as 1 check and then write available values in return?

Reply:
The returning errors are a bit difference so I think the following impl is better.